app/repositories/PublishedRepository.scala

```diff
@@ -19,17 +19,18 @@ package repositories
 import java.time.ZonedDateTime
 
 import javax.inject.{Inject, Singleton}
-import models.errors.{DatabaseError, NotFoundError}
+import models.errors.{DatabaseError, DuplicateKeyError, NotFoundError}
 import models.{PublishedProcess, RequestOutcome}
-import play.api.libs.json.{Format, JsObject, Json}
+import play.api.libs.json.{Format, JsObject, JsResultException, Json}
 import play.modules.reactivemongo.ReactiveMongoComponent
 import reactivemongo.api.indexes.{Index, IndexType}
+import reactivemongo.bson.{BSONBoolean, BSONInteger, BSONString}
```

Comment:
BSONBoolean, BSONInteger and BSONString are not used

Reply:
done